### DISCUSSION TITLE
What is the meaning of these values in segments results: avg_logprob, compression_ratio, no_speech_prob, temperature?

### DISCUSSION THREAD

**Discussion by toanhn** [SYSTRAN/faster-whisper#1042] [Q&A]
What is the meaning of these values in segments results: avg_logprob, compression_ratio, no_speech_prob, temperature?
Thanks.

**Comment by Hikari-Tsai** [comment]
They are decoding diagnostics, not four independent confidence probabilities:

- `avg_logprob`: average log probability of the generated token sequence. Values closer to zero are generally more confident, but it is not a calibrated probability and is affected by language, length and decoding settings.
- `compression_ratio`: gzip compression ratio of the decoded text. A high value often indicates repetitive output or a hallucination loop. It is used with `compression_ratio_threshold` to trigger fallback decoding.
- `no_speech_prob`: probability assigned to Whisper's no-speech token for the window. A segment is skipped only when this exceeds `no_speech_threshold` **and** `avg_logprob` is below `log_prob_threshold`; it should not be interpreted alone.
- `temperature`: the sampling temperature that produced the accepted result. The default temperature list is tried successively when log-probability or compression checks fail. `0` uses beam search; a positive value uses sampling.

For a UI confidence indicator, do not convert `exp(avg_logprob)` directly into an accuracy percentage. If `word_timestamps=True`, word-level `probability` values are more localized, but they are still uncalibrated. Calibrate any threshold against labeled audio from the target domain.
